standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HMSHandler.java

```diff
@@ -5461,6 +5462,37 @@ public GetPartitionResponse get_partition_req(GetPartitionRequest req)
    */
   private void fireReadTablePreEvent(String catName, String dbName, String tblName)
       throws MetaException, NoSuchObjectException {
+    if (catName == null) {
+      throw new NullPointerException("catName is null");
+    }
+
+    if (isBlank(catName)) {
```

Comment:
I think we can converge the null check and blank check into a single condition and throw a NoSuchObjectException in that case. We shouldn't be throwing a NullPointerException.